core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategy.java

```diff
@@ -39,45 +39,63 @@
 public interface RetryStrategy {
 
     /**
-     * A {@link RetryStrategy} that defines a retry should not be performed.
+     * Returns a {@link RetryStrategy} that never retries.
      */
     static RetryStrategy never() {
         return (ctx, cause) -> CompletableFuture.completedFuture(null);
     }
 
     /**
-     * A {@link RetryStrategy} that retries only on {@link UnprocessedRequestException} with
-     * the {@link Backoff#ofDefault()}.
+     * Returns a {@link RetryStrategy} that retries with {@link Backoff#ofDefault()}
+     * only on an {@link UnprocessedRequestException}.
      */
     static RetryStrategy onUnprocessed() {
         return onUnprocessed(Backoff.ofDefault());
     }
 
     /**
-     * A {@link RetryStrategy} that retries only on {@link UnprocessedRequestException} with the specified
-     * {@link Backoff}.
+     * Returns a {@link RetryStrategy} that retries with the specified {@link Backoff}
+     * only on an {@link UnprocessedRequestException}.
      */
     static RetryStrategy onUnprocessed(Backoff backoff) {
         requireNonNull(backoff, "backoff");
+        return onException(cause -> cause instanceof UnprocessedRequestException ? backoff : null);
+    }
+
+    /**
+     * Returns a {@link RetryStrategy} that retries with {@link Backoff#ofDefault()} on any {@link Exception}.
+     */
+    static RetryStrategy onException() {
+        return onException(cause -> Backoff.ofDefault());
+    }
+
+    /**
+     * Returns a {@link RetryStrategy} that decides to retry using the specified {@code backoffFunction}.
+     *
+     * @param backoffFunction A {@link Function} that returns the {@link Backoff} or {@code null} (no retry)
+     *                        according to the given {@link Throwable}
+     */
+    static RetryStrategy onException(Function<? super Throwable, ? extends Backoff> backoffFunction) {
+        requireNonNull(backoffFunction, "backoffFunction");
         return onStatus((status, thrown) -> {
-            if (thrown != null && Exceptions.peel(thrown) instanceof UnprocessedRequestException) {
-                return backoff;
+            if (thrown != null) {
+                return backoffFunction.apply(Exceptions.peel(thrown));
             }
             return null;
         });
     }
 
     /**
-     * Returns the {@link RetryStrategy} that retries the request with the {@link Backoff#ofDefault()}
-     * when the response status matches {@link HttpStatusClass#SERVER_ERROR} or an {@link Exception} is raised.
+     * Returns a {@link RetryStrategy} that retries with the {@link Backoff#ofDefault()}
+     * when the response status is 5xx (server error) or an {@link Exception} is raised.
```

Comment:
now that we have `onException` wouldn't it make sense to only handle server status codes here?

Reply:
I guess a user will be interested in exceptions if he/she is interested in response status, but yeah it highly depends on a user's intention. Let us handle this in #1634.